filament/src/Renderer.cpp

```diff
@@ -235,24 +235,29 @@ void FRenderer::renderJob(ArenaScope& arena, FView* view) {
     recordHighWatermark(commands);
 }
 
-void FRenderer::mirrorFrame(FSwapChain* toSwapChain, FView const* toView, FView const* fromView) {
+void FRenderer::mirrorFrame(FSwapChain* dstSwapChain, Viewport const* dstViewport, Viewport const* srcViewport) {
     SYSTRACE_CALL();
 
     assert(mSwapChain);
-    assert(toSwapChain);
+    assert(dstSwapChain);
     FEngine& engine = getEngine();
     FEngine::DriverApi& driver = engine.getDriverApi();
-    Viewport const& fromVp = fromView->getViewport();
-    Viewport const& toVp = toView->getViewport();
 
     const Handle<HwRenderTarget> viewRenderTarget = getRenderTarget();
 
-    toSwapChain->makeCurrent(driver, mSwapChain);
+    // Set the current swap chain as the read surface, and the destination
+    // swap chain as the draw surface so that blitting between default render
+    // targets results in a frame copy from the current frame to the
+    // destination.
+    driver.makeCurrent(dstSwapChain->getHwHandle(), mSwapChain->getHwHandle());
+
     driver.blit(TargetBufferFlags::COLOR,
-                viewRenderTarget, toVp.left, toVp.bottom, toVp.width, toVp.height,
-                viewRenderTarget, fromVp.left, fromVp.bottom, fromVp.width, fromVp.height);
-    toSwapChain->commit(driver);
+                viewRenderTarget, dstViewport->left, dstViewport->bottom, dstViewport->width, dstViewport->height,
+                viewRenderTarget, srcViewport->left, srcViewport->bottom, srcViewport->width, srcViewport->height);
+    dstSwapChain->commit(driver);
```

Comment:
Do we really want to do a `commit()` here? The destination swapchain is not owned by filament. A commit is a big deal -- what if the caller wanted to do additional rendering? Conceptually that swapchain could be owned by someone else, maybe even another filament instance?

I feel like it should be the responsibility of the caller to do this. This method should be just about drawing into it.

Moreover, how does this interact with `setPresentationTime()`? I think it doesn't and that's a problem. The presentation time should be set on that surface -- and again I'm not sure it is filament's duty to do that.

I'm okay if we decide to go with the `commit()` here, but it should be clearly documented in the doxygen doc.

Reply:
added flags to control whether commit and setPresentationTime are called on dstSwapChain